trac_to_markdown.py

```diff
@@ -0,0 +1,89 @@
+"""Convert Trac's WikiFormat text to Markdown."""
+
+import re
+
+
+FLAVOURS = {
+    "markdown": {
+        "bold": r"**\1**",
+        "italic": r"*\1*",
+        "ul": r"*\2",
+        "ol": r"\1.\2",
+        "wiki_link": r"[%s/wiki/\2](\1)",
+        "ticket_link": r"[%s/ticket/\2](\1)",
+        "changeset_link": r"[%s/changeset/\2](\1)",
+    },
+    # mrkdwn is Slack's markup language, which is similar to Markdown.
+    "mrkdwn": {
+        "bold": r"*\1*",
+        "italic": r"_\1_",
+        "ul": r"\u2022\2",
+        "ol": r"\1\2",
+        "wiki_link": r"<%s/wiki/\1|\2>",
+        "ticket_link": r"<%s/ticket/\1|\2>",
+        "changeset_link": r"<%s/changeset/\1|\2>",
+    },
+}
+
+
+def convert(text, base="", flavour="markdown"):
+    """Convert the passed text from WikiFormatting to Markdown."""
+    transforms = FLAVOURS[flavour]
+
+    # Convert code blocks.
+    # XXX This doesn't handle nested blocks, but those should be
+    # XXX uncommon. We would probably need to properly parse the text
+    # XXX to handle that.
+    text = re.sub(r"^{{{\s(.*?)\s}}}$", r"```\n\1\n```", text, re.DOTALL)
```

Comment:
Also I'm pretty sure that's the positional argument for `count` not `flags` in `re.sub` :) Fallen into this trap quite recently myself.